python/sdk_e2e/test_track_2/test_send_message.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+import pytest
+import logging
+import time
+from thief_constants import Flags, Fields, Commands
+from track2 import topic_builder
+
+logger = logging.getLogger(__name__)
+logger.setLevel(level=logging.INFO)
+
+
+@pytest.fixture
+def test_message(message_wrapper):
+    return message_wrapper(
+        {
+            Fields.THIEF: {
+                Fields.CMD: Commands.SEND_OPERATION_RESPONSE,
+                Fields.FLAGS: [Flags.RESPOND_IMMEDIATELY],
+            }
+        }
+    )
+
+
+@pytest.mark.describe("Device Client send_message method")
+class TestSendMessage(object):
+    @pytest.mark.it("Can send a simple message")
+    def test_send_message(self, client, test_message, c2d_subscription):
+        time.sleep(2)
+        client.publish(
+            topic_builder.build_telemetry_publish_topic(message=test_message.message),
+            test_message.message.get_payload(),
+            qos=1,
+        )
+        test_message.running_op.event.wait()
+
+    @pytest.mark.it("sends if disconnected after sending and manually reconnected")
+    def test_sends_if_disconencted(self, client, test_message, connection_status, auth, keep_alive):
+        client.disconnect()
+        connection_status.wait_for_disconnected()
+
+        client.publish(
+            topic_builder.build_telemetry_publish_topic(message=test_message.message),
+            test_message.message.get_payload(),
+            qos=1,
+        )
+
+        assert not test_message.running_op.event.is_set()
+
+        # TODO: this test fails badly if we add this sleep here.  It's not just that it fails,
```

Comment:
Why would this happen, and why is the sleep even useful for this test?

Reply:
It's useful because it shows an order that I, as customer, could call the Paho APis in a way that can break it pretty badly.  If there is any time between calls to `pubish` and `reconnect`, it's broken in a way that it really hard to detect and hard to recover from.  (Unless you really understand the internals of Paho, but that kinda defeats the point ... )

This specific bug, with the sleep, is actually a bug in Paho.  But, if our strategy relies on customers callin Paho directly, it's something that we need to figure out.